### PR TITLE
Add a CI step to avoid running the tmate session on the self-hosted machine

### PR DESCRIPTION
**Pull request by dashanji**
## What do these changes do?

As titled.

## Related issue number

Fix https://github.com/alibaba/GraphScope/issues/2528.

### CODE REVIEW

**Review by sighingnow — APPROVED** [review]
LGTM.

**Comment by dashanji** [comment]
Wait for testing here.

File: .github/workflows/gaia.yml [hunk 1]
```diff
@@ -23,6 +23,12 @@ jobs:
   gaia-test:
     runs-on: [self-hosted, manylinux2014]
     steps:
+    - name: Detect the tmate session
+      if: "jobs.gaia-test.steps[*].uses == 'mxschmitt/action-tmate@*'"
```
Comment (siyuan0322) [inline .github/workflows/gaia.yml:27]: could we use '*/action-tmate'?
Since mxschmitt is not the only one that provides action-tmate.

**Review by siyuan0322 — APPROVED** [review]

File: .github/workflows/gaia.yml (hunk 1 above)
Comment (dashanji) [reply .github/workflows/gaia.yml]: Make sense to me.

**Comment by dashanji** [comment]
From the [github steps context](https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context), we can't get the info of steps that wait for running. So use a static code check here.

File: .github/workflows/pegasus.yml [hunk 2]
```diff
@@ -50,6 +50,12 @@ jobs:
   k8s-test:
     runs-on: [self-hosted, ubuntu2004]
     steps:
+    - name: Detect the tmate session
+      if: "jobs.k8s-test.steps[*].uses == 'mxschmitt/action-tmate@*'"
```
Comment (siyuan0322) [inline .github/workflows/pegasus.yml:54]: why this is still use the context while others use `grep`

**Review by sighingnow — COMMENTED** [review]
`k8s-ci.yaml` and `local-ci.yaml`: put the check to `changes` job as others will depends on that to avoid duplication so many times..

File: .github/workflows/pegasus.yml (hunk 2 above)
Comment (dashanji) [reply .github/workflows/pegasus.yml]: My bad. Thanks for catching this.

**Comment by dashanji** [comment]
> `k8s-ci.yaml` and `local-ci.yaml`: put the check to `changes` job as others will depends on that to avoid duplication so many times..

Addressed.

**Pull request merged**